File: SpacexWebApp.Server/Data/Models/CustomLaunch.cs
using System.Text.Json.Serialization;

namespace SpacexWebApp.Server.Data.Models
{
    public class CustomLaunch
    {
        [JsonPropertyName("flight_number")]
        public int FlightNumber { get; set; }
        public string Name { get; set; }
        [JsonPropertyName("date_utc")]
        public string DateUtc { get; set; }
        [JsonPropertyName("rocket")]
        public Rocket Rocket { get; set; }
        [JsonPropertyName("success")]
        public bool? Success { get; set; }
        [JsonPropertyName("details")]
        public string Details { get; set; }
        [JsonPropertyName("links")]
        public Links Links { get; set; }
        [JsonPropertyName("crew")]
        public List<Crew> Crew { get; set; }
        [JsonPropertyName("launchpad")]
        public Launchpad Launchpad { get; set; }
        [JsonPropertyName("id")]
        public string Id { get; set; }
    }
}

File: spacexwebapp.client/src/app/shared-models/custom-launch.ts
import { Crew } from "./crew";
import { Links } from "./links";
import { Launchpad } from "./launchpad";
import { Rocket } from "./rocket";

export interface CustomLaunch {
  flight_number: number;
  name: string;
  date_utc: Date;
  rocket: Rocket;
  success: boolean | null;
  details: string;
  links: Links;
  crew: Crew[];
  launchpad: Launchpad;
  id: string;
}
